packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -153,6 +195,174 @@ export class AddressBookController extends BaseController<
 
     return true;
   }
+
+  /**
+   * Initialization step which optionally listens to changes to frequent RPC
+   * chain IDs in {@link PreferencesController}) state and ensures that
+   * addresses are filed under the correct chain.
+   */
+  #handleChainIdChanges() {
+    const { getProvider } = this.config;
+
+    if (
+      this.config.syncWithRpcChanges &&
+      this.config.onPreferencesStateChange !== undefined &&
```

Comment:
Perhaps we should throw an error if `syncWithRpcChanges` is ever set without `getProvider` or `onPreferencesStateChange` being set?